### PR TITLE
Print out `yarn install` when yarn.lock file is present

### PR DESCRIPTION
**Pull request by samdemaeyer**
As some might know, when you are using yarn and you remove `/node_modules`, you will have a comment when you try to run `ember server`
```
node_modules appears empty, you may need to run `npm install`
```
screen shot 2017-03-22 at 16 21 42

I fixed something similar in [ember-cli-dependency-checker](https://github.com/quaertym/ember-cli-dependency-checker/pull/68).
It's quite confusing for people who are not that familiar with the package managers.

As this is my first PR to ember-cli, I am guessing that there might be a better way for this fix. But at least the idea is out there now.

In this PR, I will print out 
```
node_modules appears empty, you may need to run `yarn`
```
screen shot 2017-03-22 at 16 35 18

when there is a `yarn.lock` file present.

I know that yarn is not used by ember-cli by default yet, but it might be ok to check this as in the team I am working in, it got confusing for some people .. 😅

### CODE REVIEW

File: lib/models/command.js [hunk 1]
```diff
@@ -288,7 +288,11 @@ let Command = CoreObject.extend({
 
       if (this.works === 'insideProject') {
         if (!this.project.hasDependencies()) {
-          throw new SilentError('node_modules appears empty, you may need to run `npm install`');
+          let installInstuction = '`npm install`';
+          if (this.project.hasYarnLockFile()) {
```
Comment (rwjblue) [inline lib/models/command.js]: I'd prefer to check it here directly, and not add another method to `project`. 

If however, we do keep the project method around, we should update other locations that search for `yarn.lock` to use it ([example here](https://github.com/ember-cli/ember-cli/blob/bd2822868e3b9bf11cfa9f28b91cb4d021c6680b/lib/commands/init.js#L64)).

Comment (samdemaeyer) [inline lib/models/command.js]: @rwjblue, we only use it in [/lib/commands/init.js:64](https://github.com/ember-cli/ember-cli/blob/bd2822868e3b9bf11cfa9f28b91cb4d021c6680b/lib/commands/init.js#L64) and [lib/tasks/npm-task.js:45](https://github.com/ember-cli/ember-cli/blob/bd2822868e3b9bf11cfa9f28b91cb4d021c6680b/lib/tasks/npm-task.js#L45)
I can take it out and check it directly if you like.
Than I just need another approach to test it, ATM, we overwrite the `hasYarnLockFile` function in the project class:
```js
command.project.hasYarnLockFile = function() { return true; };
```

Comment (samdemaeyer) [inline lib/models/command.js]: @rwjblue, in [this commit](https://github.com/ember-cli/ember-cli/pull/6888/commits/ea2f6dcbd96783d7e5d67cf68de38022a8eafcfc) I will check it directly. To test this behaviour, I am renaming the `ember-cli` project's actual `yarn.lock` file before the test runs and after the test, rewrite it back to the original name.
Could you advise me if this is a good approach, or might it be to risky? If so, what would you suggest for testing?

File: lib/models/command.js [hunk 2]
```diff
@@ -288,7 +289,11 @@ let Command = CoreObject.extend({
 
       if (this.works === 'insideProject') {
         if (!this.project.hasDependencies()) {
-          throw new SilentError('node_modules appears empty, you may need to run `npm install`');
+          let installInstuction = '`npm install`';
+          if (existsSync(path.join(this.project.root, 'yarn.lock'))) {
+            installInstuction = '`yarn`';
```
Comment (Turbo87) [inline lib/models/command.js]: can we make this `yarn install` to make it a bit easier to understand?

Comment (samdemaeyer) [inline lib/models/command.js]: @Turbo87, sure, no prob.

File: tests/unit/commands/generate-test.js [hunk 3]
```diff
@@ -47,9 +49,20 @@ describe('generate command', function() {
 
   it('runs GenerateFromBlueprint but with null nodeModulesPath', function() {
     command.project.hasDependencies = function() { return false; };
+    let originalYarnLockPath = path.join(command.project.root, 'yarn.lock');
+    let dummyYarnLockPath = path.join(command.project.root, 'foo.bar');
+    fs.renameSync(originalYarnLockPath, dummyYarnLockPath);
 
     return expect(command.validateAndRun(['controller', 'foo'])).to.be.rejected.then(reason => {
       expect(reason.message).to.eql('node_modules appears empty, you may need to run `npm install`');
+    }).then(() => fs.renameSync(dummyYarnLockPath, originalYarnLockPath));
```
Comment (Turbo87) [inline tests/unit/commands/generate-test.js]: this `renameSync` will not run if the assertion above fails. you could wrap it in a describe() block though:

```js
describe('without yarn.lock file', function() {
  let originalYarnLockPath, dummyYarnLockPath;

  beforeEach(function() {
    originalYarnLockPath = path.join(command.project.root, 'yarn.lock');
    dummyYarnLockPath = path.join(command.project.root, 'foo.bar');
    fs.renameSync(originalYarnLockPath, dummyYarnLockPath);
  });

  afterEach(function() {
    fs.renameSync(dummyYarnLockPath, originalYarnLockPath);
  });

  it('runs GenerateFromBlueprint but with null nodeModulesPath', function() {
    command.project.hasDependencies = function() { return false; };

    return expect(command.validateAndRun(['controller', 'foo'])).to.be.rejected.then(reason => {
      expect(reason.message).to.eql('node_modules appears empty, you may need to run `npm install`');
    });
  });
});
``

Comment (samdemaeyer) [inline tests/unit/commands/generate-test.js]: Thanks @Turbo87, makes sense!
Question, would you typically destroy the global variables in the `afterEach` function?
```js
afterEach(function() {
  fs.renameSync(dummyYarnLockPath, originalYarnLockPath);
  originalYarnLockPath = dummyYarnLockPath = null;
});
```

Comment (Turbo87) [inline tests/unit/commands/generate-test.js]: since they are reassigned in every `beforeEach()` block there is IMHO no need to reset them in `afterEach()`

**Comment by Turbo87** [comment]
@homu r+

**Pull request merged**